src/Disciplina/Educator/Validate.hs

```diff
@@ -0,0 +1,32 @@
+
+-- | Validation of private block
+
+module Disciplina.Educator.Validate
+       ( validate
+       ) where
+
+import Universum
+
+import Disciplina.Educator.Block (PrivateBlock, pbHeader, pbBody, pbbTxs, pbhBodyProof)
+import Disciplina.Educator.Txs (PrivateTxAux (..), PrivateTxWitness (..))
+import Disciplina.Educator.Serialise ()
+import Disciplina.Crypto.MerkleTree (fromFoldable, getMerkleRoot)
+import Disciplina.Crypto (hash, verify)
+
+-- | Validate private block.
+-- Block is valid if header matches body
+-- and all transactions have valid signatures.
+validate :: PrivateBlock -> Bool
+validate pb =
+    let txs = pb ^. pbBody . pbbTxs
+        merkleRoot = getMerkleRoot (fromFoldable txs)
+        headerProof = pb ^. pbHeader . pbhBodyProof
+        headerProofValid = headerProof == merkleRoot
+        txsSigProofValid = all (==True) (map validateTx txs)
+    in headerProofValid && txsSigProofValid
```

Comment:
This is all good, but in real life you'll probably want to at least log some error messages if some checks failed. In order to provide some info about failure if validation has failed, you may use `VerificationRes` datatype from `serokell-util` instead of `Bool`: https://github.com/serokell/serokell-util/blob/master/src/Serokell/Util/Verify.hs